MigrationPreparer/MigrationPreparer.json

```diff
@@ -0,0 +1,335 @@
+            ]
+        },
+        "getComponentReadiness": {
+            "summary": "Returns the list of components which are already ready for migration",
+            "result": {
+                "type": "object",
+                "properties": {
+                    "componentList": {
+                        "type": "array",
+                        "items": {
+                            "$ref": "#/definitions/componentName"
+                        }
+                    },
+                    "success": {
+                        "$ref": "#/definitions/success"
+                    }
+                },
+                "required": [
+                    "componentList",
+                    "success"
+                ]
+            },
+            "errors": [
+                {
+                    "description": "Unknown error",
+                    "properties": {
+                        "code": {
+                            "type": "string",
+                            "description": "Error code"
+                        },
+                        "message": {
+                            "type": "string",
+                            "description": "Error message"
+                        }
+                    },
+                    "required": [
+                        "code",
+                        "message"
+                    ]                    
+                }
+            ]
+        },
+        "reset": {
+            "summary": "Reset based on resettype, should delete all entries in the datastore or componentList for migration or both",
+            "params": {
+                "type": "object",
+                "properties": {
+                    "resetType": {
+                        "$ref": "#/definitions/resetType"
+                    }
+                },
+                "required": [
+                    "resetType"
+                ]
+            },
+            "result": {
+                "type": "object",
+                "properties": {
+                    "success": {
+                        "$ref": "#/definitions/success"
+                    }
+                },
+                "required": [
+                    "success"
+                ]
+            },
+            "errors": [
+                {
+                    "description": "Unknown error",
+                    "properties": {
+                        "code": {
+                            "type": "string",
+                            "description": "Error code"
+                        },
+                        "message": {
+                            "type": "string",
+                            "description": "Error message"
+                        }
+                    },
+                    "required": [
+                        "code",
+                        "message"
+                    ]                    
+                }
+            ]
+        }
+    },
+    "events": {
+        "onValueChanged": {
```

Comment:
Please remove this. This plugin does not offer "value changed" notification.

Reply:
"value changed" notification was kept dummy just in case for any future purpose, will that too...